File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        ios_base::sync_with_stdio(false);
        cin.tie(nullptr);
        cout.tie(nullptr);
        if(n<=0)
            return 0;
        if(n==1)
            return 1;
        int a{1},b{1};

        for(int i=1;i<n;i++)
        {
            int temp = a;
            a = b;
            b = temp + b;
        }
        return b;
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 0:
            return 0
        elif n == 1:
            return 1

        fib_0 = 0
        fib_1 = 1

        for i in range(2, n + 2):
            fib_n = fib_0 + fib_1
            fib_0, fib_1 = fib_1, fib_n

        return fib_1